### PR TITLE
[Review Needed]Pytest Style analysis/test_waterdynamics.py

### PR DESCRIPTION
**Pull request by utkbansal**
Fixes #

Changes made in this Pull Request:
 - 

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [ ] Issue raised/referenced?

### CODE REVIEW

File: testsuite/MDAnalysisTests/analysis/test_waterdynamics.py [hunk 1]
```diff
@@ -22,64 +22,81 @@
...
+    return MDAnalysis.Universe(waterPSF, waterDCD)
+
+
+def test_HydrogenBondLifetimes(universe):
+    hbl = MDAnalysis.analysis.waterdynamics.HydrogenBondLifetimes(universe,
+                                                                  SELECTION1,
+                                                                  SELECTION2,
+                                                                  0, 5, 3)
+    hbl.run(verbose=False)
+    assert round(hbl.timeseries[2][1], 5) == 0.75
```
Comment (jbarnoud) [inline testsuite/MDAnalysisTests/analysis/test_waterdynamics.py]: `assert_almost_equal`?

File: testsuite/MDAnalysisTests/analysis/test_waterdynamics.py [hunk 2]
```diff
@@ -22,64 +22,81 @@
...
-        sp.run(verbose=False)
-        assert_equal(round(sp.timeseries[1],5), 1.0)
-    
-    def test_SurvivalProbability_zeroMolecules(self):
-        sp_zero = MDAnalysis.analysis.waterdynamics.SurvivalProbability(self.universe, 
-                                                    self.selection3, 0, 6, 3)
-        sp_zero.run(verbose=False)
-        assert_equal(sp_zero.timeseries[1], 0.0)
+SELECTION1 = "byres name OH2"
+SELECTION2 = SELECTION1
```
Comment (jbarnoud) [inline testsuite/MDAnalysisTests/analysis/test_waterdynamics.py]: I do not see a reason to have the `SELECTION2` constant. `SELECTION1` can be used instead just as well.

File: testsuite/MDAnalysisTests/analysis/test_waterdynamics.py (hunk 1 above)
Comment (utkbansal) [inline testsuite/MDAnalysisTests/analysis/test_waterdynamics.py]: @jbarnoud I think it the exact equal works then we don't need to do almost equals!?

Comment (jbarnoud) [inline testsuite/MDAnalysisTests/analysis/test_waterdynamics.py]: It looks like the test tries to compare `hbl.timeseries[2][1]` to 0.75 but has precision issues. This is what `assert_almost equal` is for.

Comment (utkbansal) [inline testsuite/MDAnalysisTests/analysis/test_waterdynamics.py]: Ok.

File: testsuite/MDAnalysisTests/analysis/test_waterdynamics.py [hunk 3]
```diff
@@ -22,64 +22,80 @@
...
+                                                                   0, 10, 2)
+    msd.run(verbose=False)
+    assert round(msd.timeseries[1], 5) == 0.03984
+
+
+def test_MeanSquareDisplacement_zeroMolecules(universe):
+    msd_zero = MDAnalysis.analysis.waterdynamics.MeanSquareDisplacement(
+        universe,
+        SELECTION2, 0, 10, 2)
+    msd_zero.run(verbose=False)
```
Comment (kain88-de) [inline testsuite/MDAnalysisTests/analysis/test_waterdynamics.py]: adding the `verbose=False` is not necessary. That already is the default.

**Pull request merged**